components/server/ee/src/billing/billing-mode.ts

```diff
@@ -0,0 +1,121 @@
+    @inject(SubscriptionService) protected readonly subscriptionSvc: SubscriptionService;
+    @inject(StripeService) protected readonly stripeSvc: StripeService;
+    @inject(TeamSubscription2DB) protected readonly teamSubscription2Db: TeamSubscription2DB;
+
+    async getBillingModeForUser(user: User): Promise<BillingMode> {
+        if (!this.config.enablePayment) {
+            // Payment is not enabled. E.g. Self-Hosted.
+            return { mode: "none" };
+        }
+        const now = new Date();
+
+        // Is Usage Based Billing enabled for this user or not?
+        const isUsageBasedBillingEnabled = await getExperimentsClientForBackend().getValueAsync(
+            "isUsageBasedBillingEnabled",
+            false,
+            {
+                user,
+            },
+        );
+
+        // Any Chargebee subscriptions? This includes those derived from a Team Seat!
+        const cbSubscriptions = await this.subscriptionSvc.getActivePaidSubscription(user.id, now);
+        if (cbSubscriptions.length > 0) {
+            if (
+                isUsageBasedBillingEnabled &&
+                cbSubscriptions.every((s) => Subscription.isCancelled(s, now.toISOString()))
+            ) {
+                // The user has one or more paid subscriptions, but all of them have already been cancelled
+                return { mode: "chargebee", tier: "paid_cancelled_and_ubb" };
+            }
+
+            // The user has at least one paid subscription
+            return { mode: "chargebee", tier: "paid", planId: cbSubscriptions[0].planId! };
+        }
+
+        // UBB enabled?
+        if (!isUsageBasedBillingEnabled) {
+            // No Chargebee subscription, and UBB is not enabled: CB free tier?
+            return { mode: "chargebee", tier: "free" };
+        }
+
+        // Stripe: Active subsciption?
+        const customer = await this.stripeSvc.findCustomerByUserId(user.id);
+        if (customer) {
+            const subscription = await this.stripeSvc.findUncancelledSubscriptionByCustomer(customer.id);
+            if (subscription) {
+                return { mode: "usage-based", tier: "paid" };
+            }
+        }
+
+        // Team membership is not relevant here, because we're interested in the user perspective
+        return { mode: "usage-based", tier: "free" };
+    }
+
+    async getBillingModeForTeam(team: Team): Promise<BillingMode> {
+        if (!this.config.enablePayment) {
+            // Payment is not enabled. E.g. Self-Hosted.
+            return { mode: "none" };
+        }
+        const now = new Date().toISOString();
+
+        // Is Usage Based Billing enabled for this team?
+        const isUsageBasedBillingEnabled = await getExperimentsClientForBackend().getValueAsync(
+            "isUsageBasedBillingEnabled",
+            false,
+            {
+                teamId: team.id,
+                teamName: team.name,
+            },
+        );
```

Comment:
If I'm not mistaken, we don't yet have anything like this in ConfigCat -- we only enable UBB for users who happen to be part of the Gitpod team, and then everywhere in the UI (i.e. for all their teams).

But maybe it could be a good idea to selectively enable UBB for a specific team, and only in that team's UI 👍

Reply:
> But maybe it could be a good idea to selectively enable UBB for a specific team, and only in that team's UI +1 we'd just need to code this into ConfigCat.

That was the intend. `teamId` and `team_name` are already available in ConfigCat. :+1: